public/app/plugins/datasource/cloud-monitoring/datasource.ts

```diff
@@ -176,8 +114,29 @@ export default class CloudMonitoringDatasource extends DataSourceApi<CloudMonito
         },
       ],
       range: this.timeSrv.timeRange(),
-    } as DataQueryRequest<CloudMonitoringQuery>)
+    } as DataQueryRequest<CloudMonitoringQuery>;
+
+    const queries = options.targets
+      .map(this.migrateQuery)
+      .filter(this.filterQuery)
+      .map(q => this.applyTemplateVariables(q, options.scopedVars));
```

Comment:
I know getLabels used to call getTimeSeries and that function used to call  migrateQuery, filterQUery and applyTemplateVariables before. But that might not be necessary now. The getLabels query is already prepared on line 101, so I think you should be able to remove line 119-122.